Bindings for librsvg

PaulJohnson opened the pull request:
This is a binding for the SVG rendering library. It lets you include an SVG file as part of a Cairo render.

It also includes some updates to the bindings/README that should hopefully help other creators of bindings in the future.

PaulJohnson commented:
One gotcha: this depends on gi-cairo-render, which in turn has an upper bound of 0.24 on the haskell-gi-base package. So you need to use `--allow-newer` to build until that gets fixed.

garetxe commented:
Thanks a lot! I would be happy to merge after some minor changes:

+ Could you please add the relevant ubuntu package name in `pkg.info`, so that CI installs it, and CI passes? You need to add the package that provides `librsvg-2.0>=2.50` in ubuntu, probably [librsvg2-dev](https://packages.ubuntu.com/focal/librsvg2-dev). Note that the version of the package in ubuntu 20.04, which is what the CI runs under, seems to be 2.48, so you might want to relax the bound.

+ I am a little puzzled about the comments about the haddocks: the function prototype (which goes into the haddocks) is generated out of the introspection data in your system. It is a common problem that some nullable annotations are missing here, but running the xlstproc thing should not do anything if you are just running locally: it simply makes explicit as an override what is in your installed gir file, so it should have no effect if you run it in the same system where you generated the overrides with `Nullable.xlst` (the use is mostly to add modern fixed annotations to an old library, whose old versions did not have such annotations: you run `xsltproc` in a new system with the fixed introspection data and then people building the bindings on old systems with missing nullable annotations get them from the overrides. So it is a form of "backporting" nullable annotations, if you like.). Could you please elaborate on what you saw here, and how running `xlstproc` fixes things?

Thanks again!

PaulJohnson commented:
I've changed `pkg.info` as requested. I've also changed the additional text in `bindings/README.md` based on your reply. If it still isn't right then it might be better to edit the text yourself; I'm still feeling my way here.

PaulJohnson commented:
I've fixed the immediate cause of the build fail, but it still won't build because of a version clash with gi-cairo-render, which requires `haskell-gi-base>=0.24 && <0.25`.

PaulJohnson commented:
Nope, still failing to build. I could remove example/rsvg from the build, but that's not really the solution either. How would you like to proceed with this?

garetxe commented:
Thanks for the changes! The new PR looks basically ready to merge to me, but I would prefer to keep CI working. The ideal solution would be to fix `gi-cairo-render` to relax the bounds, so in order to help this happen I have reported issue https://github.com/cohomology/gi-cairo-render/issues/7 (which seems to me to be the best long-term approach here).

Hopefully that's work, and then we can merge the PR as is! I would rather keep the example around.

garetxe commented:
I merged `gi-cairo-render` into this repo, so I think that this is now ready to merge. Thanks again for the bindings!

Would you like me to upload `gi-rsvg` to hackage?

PaulJohnson commented:
If you're asking me, then yes please. 

On 1 August 2021 08:08:48 BST, "Iñaki" ***@***.***> wrote:
>I merged `gi-cairo-render` into this repo, so I think that this is now ready to merge. Thanks again for the bindings!
>
>Would you like me to upload `gi-rsvg` to hackage?
>
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/haskell-gi/haskell-gi/pull/354#issuecomment-890465157
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

garetxe commented:
Done, sorry it took a little while!
